### PR TITLE
DAOS-9201 API: Remove compatibility code for daos_cont_create()

### PR DESCRIPTION
**Pull request by wiliamhuang**
Port the changes of removing compatibility code for daos_cont_create() to 2.2.

Features: container
Test-tag: pr container dm_dst_create dm_obj_small basiccheckout_dm

Signed-off-by: Lei Huang <lei.huang@intel.com>

### CODE REVIEW

**Review by daltonbohning — COMMENTED** [review]
Is this a cherry-pick of #7834 / [5501662](https://github.com/daos-stack/daos/commit/55016625a45d2a0d83f8abcf45d315841b4a9e80) ? Just checking, since the commit title would be `DAOS-9201 API: Remove compatibility code for daos_cont_create() (#7834)` if `git cherry-pick` was used.

**Comment by wiliamhuang** [comment]
> Is this a cherry-pick of #7834 / [5501662](https://github.com/daos-stack/daos/commit/55016625a45d2a0d83f8abcf45d315841b4a9e80) ? Just checking, since the commit title would be `DAOS-9201 API: Remove compatibility code for daos_cont_create() (#7834)` if `git cherry-pick` was used.

I did not use "git cherry-pick". Shilong told me I should try it. I manually ported the patch and double-checked.

**Comment by wiliamhuang** [comment]
> Is this a cherry-pick of #7834 / [5501662](https://github.com/daos-stack/daos/commit/55016625a45d2a0d83f8abcf45d315841b4a9e80) ? Just checking, since the commit title would be `DAOS-9201 API: Remove compatibility code for daos_cont_create() (#7834)` if `git cherry-pick` was used.

I did not use "git cherry-pick". Shilong told me I should try it. I manually ported the patch and double-checked.

File: src/client/dfs/duns.c [hunk 1]
```diff
@@ -676,8 +676,6 @@ create_cont(daos_handle_t poh, struct duns_attr_t *attrp, bool create_with_label
 		if (create_with_label)
 			rc = dfs_cont_create_with_label(poh, attrp->da_cont, &dfs_attr,
 							&attrp->da_cuuid, NULL, NULL);
-		else if (!uuid_is_null(attrp->da_cuuid))
```
Comment (wangshilong) [inline src/client/dfs/duns.c:679]: Did you include some extra fixes to this pr, eg:
https://github.com/daos-stack/daos/pull/9364/files

For backport, normally we don't do this, we just keep as much as possible with master, and port extra
fixes one by one to release/2.2 branch eg.

**Review by wangshilong — APPROVED** [review]

File: src/client/dfs/duns.c (hunk 1 above)
Comment (wiliamhuang) [reply src/client/dfs/duns.c:679]: Thanks! There is no issue (PR 9364) in 2.2 as in master. I will port patches one by one to 2.2 in future.

**Review by johannlombardi — APPROVED** [review]

**Pull request merged**